### PR TITLE
#7594: Make FaultTolerance 2.0 work with Metrics 2.0

### PR DESCRIPTION
**Pull request by dialaya**
Submit changes
Signed-off-by: bakary dialaya djiba <dialaya@gmail.com>

### CODE REVIEW

**Comment by Azquelt** [comment]
This is for #7594 (github doesn't make a link from the number in the PR title)

**Comment by Azquelt** [comment]
Hi @dialaya, before we can merge this, we need you to sign a CLA. Could you print, sign and scan/photograph https://github.com/OpenLiberty/open-liberty/blob/master/cla/open-liberty-cla-individual.pdf, and send it to cla@openliberty.io?

I'll do a proper review of the code tomorrow, I think there'll be a couple of bits we missed. You're welcome to make those changes or I can ask someone else in my team to do them if you prefer.

Thanks, it was great to work with you on Saturday :smile:

**Comment by dialaya** [comment]
> Hi @dialaya, before we can merge this, we need you to sign a CLA. Could you print, sign and scan/photograph https://github.com/OpenLiberty/open-liberty/blob/master/cla/open-liberty-cla-individual.pdf, and send it to [cla@openliberty.io](mailto:cla@openliberty.io)?
> 
> I'll do a proper review of the code tomorrow, I think there'll be a couple of bits we missed. You're welcome to make those changes or I can ask someone else in my team to do them if you prefer.
> 
> Thanks, it was great to work with you on Saturday 😄

Hi @Azquelt ,
I will sign and send back the document.
I can make the changes. just let me know, please.

It was a pleasure for me to work with you on Saturday.

Thanks you.

**Comment by Azquelt** [comment]
#build

**Comment by Azquelt** [comment]
Thanks for the updates. Most of the comments are addressed. I tried to run a personal build to verify that it now compiles and passes the tests but it looks like that'll have to wait until the CLA is returned and processed.

**Comment by Azquelt** [comment]
Hi @dialaya, have you sent in the CLA? If so I'll chase someone to find out why the `CLA Needed` tag is still on this PR.

**Comment by Azquelt** [comment]
@LibbyBot are you kidding me? A comment prompts you to recheck?

Oh well... #build

File: dev/com.ibm.websphere.appserver.features/visibility/auto/com.ibm.websphere.appserver.mpFaultTolerance-metrics2.0.feature [hunk 1]
```diff
@@ -0,0 +1,9 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=com.ibm.websphere.appserver.mpFaultTolerance-metrics2.0
+singleton=true
+IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=com.ibm.websphere.appserver.mpFaultTolerance-1.1)(osgi.identity=com.ibm.websphere.appserver.mpFaultTolerance-2.0)))", \
+ osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=com.ibm.websphere.appserver.mpMetrics-2.0))"
+IBM-Install-Policy: when-satisfied
+-bundles=com.ibm.ws.microprofile.faulttolerance.metrics,com.ibm.ws.microprofile.faulttolerance.metrics.2.0
+kind=noship
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/com.ibm.websphere.appserver.mpFaultTolerance-metrics2.0.feature]: shouldn't this be beta?

Comment (Azquelt) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/com.ibm.websphere.appserver.mpFaultTolerance-metrics2.0.feature]: not yet

**Comment by Azquelt** [comment]
#build

**Comment by Azquelt** [comment]
Sorry this has been slow, we've been really busy working on the reactive streams feature and this slipped off my radar. I hope to get this merged once we have a good build.

**Comment by Azquelt** [comment]
Personal build failed due to an infrastructure issue.

#build

**Comment by Azquelt** [comment]
#build

**Comment by Azquelt** [comment]
#build

**Comment by Azquelt** [comment]
#build

**Comment by Azquelt** [comment]
I think the failures in the build are being caused by this being an auto-feature which is unreleased but which depends on two features which have been released. The autofeature gets pulled automatically into the release package and then causes a validation failure because it shouldn't be released and doesn't contain the correct metadata.

I'll work around this by making it depend on an extra feature which will be removed when it's ready to be released.

**Comment by Azquelt** [comment]
#build

**Pull request merged**